File: game_of_life/mrugger/java/WriterConsole.java


class WriterConsole implements IWriteable
{
  private Pair _window_size;


  public WriterConsole(Pair window_size)
  {
    _window_size = window_size;
  }


  // IWriteable interface

  public Pair begin_board()
  {
    System.out.print("\033[H");
    return _window_size;
  }

  public void begin_row()
  {}

  public void write_cell(CellType cellvalue)
  {
    System.out.print(cellvalue.to_char());//(cellvalue == CellType.CELL_ALIVE ? '*' : ' ');
  }

  public void end_row()
  {
    System.out.println();
  }

  public void end_board()
  {}
}

File: game_of_life/mrugger/ruby/writer.console.rb
require './game.types.rb'


class WriterConsole

  private

  @output_size


  public

  def initialize(output_size)
    @output_size = output_size
  end


  # Game writer interface

  def begin_board
    puts "\033[H"
    @output_size
  end

  def begin_row
  end

  def write_cell(celltype)
    putc "#{celltype == CellType::CELL_ALIVE ? '*' : ' '}"
  end

  def end_row
    puts ""
  end

  def end_board
  end

end
